scripts/builtin/steplm.dml

```diff
@@ -0,0 +1,320 @@
+    } else {
+      beta = 0.0;
+      AIC_best = n * log(sum(y ^ 2) / n);
+    }
+
+    AICs = matrix(AIC_best, rows = 1, cols = m_orig);
+    print("Best AIC without any features: " + AIC_best);
+
+    boa_ncol = ncol(X_orig);
+    if (icpt != 0) {
+      boa_ncol = boa_ncol + 1
+    }
+
+    beta_out_all = matrix(0, rows = boa_ncol, cols = m_orig * 1);
+
+    y_ncol = 1;
+    column_best = 0;
+    # First pass to examine single features
+    for (i in 1:m_orig, check = 0) {
+      columns_fixed_ordered_1 = matrix(i, rows = 1, cols = 1);
+      [AIC_1, beta_out_i] = linear_regression(X_orig[, i], y, icpt);
+
+      print(" mibeta "  + toString(beta_out_i))
+      AICs[1, i] = AIC_1;
+      AIC_cur = as.scalar(AICs[1, i]);
+      if ((AIC_cur < AIC_best) & ((AIC_best - AIC_cur) > abs(thr * AIC_best))) {
+        column_best = i;
+        AIC_best = as.scalar(AICs[1, i]);
+      }
+
+      beta_out_all[1:nrow(beta_out_i), (i - 1) * y_ncol + 1:i * y_ncol] = beta_out_i[, 1:1];
+
+    }
+
+    # beta best so far
+    beta_best = beta_out_all[, (column_best - 1) * y_ncol + 1:column_best * y_ncol];
+
+    if (column_best == 0) {
+      print("AIC of an empty model is " + AIC_best + " and adding no feature achieves more than " +
+           (thr * 100) + "% decrease in AIC!");
+      Selected = matrix(0, rows = 1, cols = 1);
+      if (icpt == 0) {
+        B = matrix(beta, rows = m_orig, cols = 1);
+      } else {
+        B_tmp = matrix(0, rows = m_orig + 1, cols = 1);
+        B_tmp[m_orig + 1,] = beta;
+        B = B_tmp;
+      }
+
+      beta_out = B;
+
+      print("estoy en el stop");
+      print(toString(beta_out) + " mi beta out");
+      print("Selected " + toString(Selected));
+
+      write(Selected, $4);
+      write(beta_out, $3);
+   
+      stop("");
+    }
+    print("Best AIC " + AIC_best + " achieved with feature: " + column_best);
+    columns_fixed[1, column_best] = 1;
+    columns_fixed_ordered[1, 1] = column_best;
+    X_global = X_orig[, column_best];
+
+    while (continue) {
+      # Subsequent passes over the features
+      beta_out_all_2 = matrix(0, rows = boa_ncol, cols = m_orig * 1);
+
+      for (i in 1:m_orig, check = 0) {
+        if (as.scalar(columns_fixed[1, i]) == 0) {
+
+          # Construct the feature matrix
+          X = cbind(X_global, X_orig[, i]);
+
+          tmp = matrix(0, rows = 1, cols = 1);
+          tmp[1, 1] = i;
+          columns_fixed_ordered_2 = append(columns_fixed_ordered, tmp);
+      
+          [AIC_2, beta_out_i] = linear_regression(X, y, icpt);
+
+          beta_out_all_2[1:nrow(beta_out_i), (i - 1) * y_ncol + 1:i * y_ncol] = beta_out_i[, 1:1];
+
+          AICs[1, i] = AIC_2;
+
+          AIC_cur = as.scalar(AICs[1, i]);
+        if ((AIC_cur < AIC_best) & ((AIC_best - AIC_cur) > abs(thr * AIC_best)) &
+            (as.scalar(columns_fixed[1, i]) == 0)) {
+          column_best = i;
+          AIC_best = as.scalar(AICs[1, i]);
+        }
+        }
```

Comment:
Here seems to be some indentation issue - please make a full pass over the script (which was not good in that regard) and make the formatting consistent. I'd recommend to enable whitespace characters in your IDE , then you'll see it at a first glance.